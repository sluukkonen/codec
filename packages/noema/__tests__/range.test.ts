import { integer, range } from "../src/index.js"
import { expectParseFailure, expectParseSuccess } from "./helpers.js"

describe("min", () => {
  test("should parse if value is less than or equal to the minimum", () => {
    const binary = range(integer, 0, 1)
    expectParseSuccess(binary, 0)
    expectParseSuccess(binary, 1)
  })

  test("should fail to parse if value is outside the bounds", () => {
    const binary = range(integer, 0, 1)
    expectParseFailure(binary, 2, [{ code: "too_large", path: [], max: 1 }])
    expectParseFailure(binary, -1, [{ code: "too_small", path: [], min: 0 }])
  })

  test("should fail to parse if value isn't parsed by the underlying codec", () => {
    const binary = range(integer, 0, 1)
    expectParseFailure(binary, "", [{ code: "invalid_integer", path: [] }])
  })
})
