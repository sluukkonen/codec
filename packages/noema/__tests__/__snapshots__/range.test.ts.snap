// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`min should fail to parse if value is outside the bounds 1`] = `
[
  {
    "code": "too_large",
    "max": 1,
    "path": [],
  },
]
`;

exports[`min should fail to parse if value is outside the bounds 2`] = `
[
  {
    "code": "too_small",
    "min": 0,
    "path": [],
  },
]
`;

exports[`min should fail to parse if value isn't parsed by the underlying codec 1`] = `
[
  {
    "code": "invalid_integer",
    "path": [],
  },
]
`;
